api/src/test/java/org/apache/iceberg/variants/VariantTestUtil.java

```diff
@@ -107,6 +130,124 @@ static SerializedPrimitive createString(String string) {
     return SerializedPrimitive.from(buffer, buffer.get(0));
   }
 
+  public static SerializedArray createMixedArray() {
+    ByteBuffer nestedBuffer = VariantTestUtil.createArray(A, C, D);
```

Comment:
I think VariantTestUtil is just a util class with the input provided externally. We don't want to provide a hard coded createINt16() e.g., here since in the test, how can you check and compare the expected value.

Better to move to the test itself.